bleak/backends/winrt/client.py

```diff
@@ -0,0 +1,835 @@
+            raise BleakError("Descriptor with handle {0} was not found!".format(handle))
+
+        read_result = await descriptor.obj.read_value_async(
+            BluetoothCacheMode.CACHED if use_cached else BluetoothCacheMode.UNCACHED
+        )
+
+        if read_result.status == GattCommunicationStatus.SUCCESS:
+            reader = DataReader.from_buffer(read_result.value)
+            # TODO: Figure out how to use read_bytes instead...
+            value = bytearray(
+                [reader.read_byte() for _ in range(reader.unconsumed_buffer_length)]
+            )
+            logger.debug("Read Descriptor {0} : {1}".format(handle, value))
+        else:
+            if read_result.status == GattCommunicationStatus.PROTOCOL_ERROR:
+                raise BleakDotNetTaskError(
+                    "Could not get GATT characteristics for {0}: {1} (Error: 0x{2:02X}: {3})".format(
+                        descriptor.uuid,
+                        _communication_statues.get(read_result.status, ""),
+                        read_result.protocol_error,
+                        CONTROLLER_ERROR_CODES.get(
+                            read_result.protocol_error, "Unknown"
+                        ),
+                    )
+                )
+            else:
+                raise BleakError(
+                    "Could not read Descriptor value for {0}: {1}".format(
+                        descriptor.uuid,
+                        _communication_statues.get(read_result.status, ""),
+                    )
+                )
+
+        return value
+
+    async def write_gatt_char(
+        self,
+        char_specifier: Union[BleakGATTCharacteristic, int, str, uuid.UUID],
+        data: bytearray,
+        response: bool = False,
+    ) -> None:
+        """Perform a write operation of the specified GATT characteristic.
+
+        Args:
+            char_specifier (BleakGATTCharacteristic, int, str or UUID): The characteristic to write
+                to, specified by either integer handle, UUID or directly by the
+                BleakGATTCharacteristic object representing it.
+            data (bytes or bytearray): The data to send.
+            response (bool): If write-with-response operation should be done. Defaults to `False`.
+
+        """
+        if not isinstance(char_specifier, BleakGATTCharacteristic):
+            characteristic = self.services.get_characteristic(char_specifier)
+        else:
+            characteristic = char_specifier
+        if not characteristic:
+            raise BleakError("Characteristic {} was not found!".format(char_specifier))
+
+        writer = DataWriter()
+        writer.write_bytes(list(data))
+        response = (
+            GattWriteOption.WRITE_WITH_RESPONSE
+            if response
+            else GattWriteOption.WRITE_WITHOUT_RESPONSE
+        )
+        write_result = await characteristic.obj.write_value_with_result_async(
+            writer.detach_buffer(), response
```

Comment:
```python
CryptographicBuffer.create_from_byte_array(list(data))
```
instead of writer

Reply:
The speed of this is not as critical, but it is almost twice as fast as the DataWriter solution. Nice!